### ISSUE TITLE
The data couldn't be read because it isn't in the correct format.

### ISSUE THREAD

**Issue by ted1ore1ee** [rrroyal/Harbour#17]
Env:
 - iOS16.1
 - Portainer installed from TrueNAS scale truecharts app
 
Step:
when I try to login with url and token, says **The data couldn't be read because it isn't in the correct format.** error, surely the URL and token are correct, what is this error mean?

Screenshot:
![IMG_9880](https://user-images.githubusercontent.com/43397442/198194620-7cef7171-a019-4898-9f2a-5d81afad796c.PNG)

**Comment by rrroyal** [comment]
please fill out the bug report template

**Comment by ted1ore1ee** [comment]
> 

thanks for your replay and I already updated the bug report as template

**Comment by blockiyt** [comment]
same here 👋🏼

**Comment by blockiyt** [comment]
I think it could be something with the port, because I run portainer behind an reverse proxy, and if I connect over the external domain, without an port, it works fine

**Comment by fmnamado** [comment]
I just came here to give you my experience.
The app got a bug that I can't remove servers. I already sent feedback related to this.
When I resintalled, obviously I hade to re-add servers and I got this bug.

My server URL is
http://192.168.xx.xx:9000
and token prt_xxxxxx

When I inserted this i got the described bug.
But I saw in portainer, that portainer itself uses inside if the container the port 9443 so I tried in harbour the following address
http://192.168.xx.xx:9443

It gave me "cannot connect" or "wrong port" or whatever message of an expected error.

But when I retried the correct 9000 port
http://192.168.xx.xx:9000

Now harbour connected!!!
I tried this for my 2 different portainer servers with the same workaround.

So I leave here the info so can help someone.

**Comment by ted1ore1ee** [comment]
> 

I tried your way this time I get invalid certificate error rather than **The data couldn't be read because it isn't in the correct format**, I think this work around actually working but since I don't have a valid certificate I still can not log in though

**Comment by rrroyal** [comment]
this should be fixed in 3.0.0 - could you all check if the issue still persists?

**Comment by peanutlasko** [comment]
I'm still having same issue. My portainer is exposed to the web via Reverse proxy. I can log in via the web app just fine, but Harbour shows error as before "The data couldn't be read because it isn't in the correct format"

My assumption is the app "defaults" to looking on port 9443 (https), but my instance of Portainer is only listening on port 9000 (http)

My reverse proxy takes care of the http to https redirection. What about just adding some "advanced setting" where the user can specify what port they want to connect with, instead of assuming 9443?

**Comment by peanutlasko** [comment]
Well the issue is that I have dozens of services that are hosted behind a Reverse Proxy - Traefik manages these and only accepts https traffic (it redirects 80 to 443 automatically) 

If I enter portainer.xxxxx.com, Traefik automatically knows to route to port 9000 (where Portainer is). I'm not able to do portainer.xxxxx.com:9000 because Traefik doesnt listen on port 9000. 

The issue here is that Portainer uses 9000 for http and 9443 for https, so I might just be stuck. Because Harbour might be assuming im coming in on https, but after my reverse proxy tunnels the traffic to port 9000, Portainer is expecting http traffic.

**Comment by rrroyal** [comment]
I don't think that Harbour assumes anything - if you specify `http://` it will expect unencrypted (and vice versa for `https`). I would need to investigate - if you'd like me to, feel free to shoot me an email with the address of the server.

**Comment by tediorelee** [comment]
> 

Hi sorry for my late response, tested on latest version of Harbour app and still getting this error
![IMG_1843](https://user-images.githubusercontent.com/48272067/212302101-1752c1a6-6672-4067-b9dc-6e48f18cd8d2.PNG)
![image](https://user-images.githubusercontent.com/48272067/212302193-f6740b54-168d-41d3-8b35-4fe401c6683f.png)
